pkg/kubeclient/client.go

```diff
@@ -0,0 +1,89 @@
+package kubeclient
+
+import (
+	"context"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+// Interface defines capability of the Kubernetes Client.
+type Interface interface {
+	// Create creates given object in Kubernetes cluster
+	Create(ctx context.Context, obj runtime.Object) error
+
+	// Get retrieves given object from Kubernetes cluster
+	Get(ctx context.Context, obj runtime.Object) error
+
+	// Update updates given object in Kubernetes cluster
+	Update(ctx context.Context, obj runtime.Object) error
+
+	// Delete deletes given object from Kubernetes cluster
+	Delete(ctx context.Context, obj runtime.Object, opts *metav1.DeleteOptions) error
+
+	// List retrieves list of object
+	List(ctx context.Context, obj runtime.Object, opts *metav1.ListOptions) error
+
+	// TODO: define capabilities for ListDeletion, Patch and Watch ?
+}
+
+// NewFromConfigOrCluster returns a Kubernetes client. If configPath is
+// provided, it uses the configPath to initialize a client else it assumes to be
+// running within a Kubernetes cluster and tries to initialize with cluster
+// configuration.
+func FromConfigOrCluster(configPath string) (Interface, error) {
+
+}
+
+// NewFromConfig returns a Kubernetes client given a path to the config file to
+// connect to Kubernetes cluster.
+func FromConfig(configPath string) (Interface, error) {
+
+}
+
+// InCluster() uses default cluster configuration to initialize a Kubernetes
+// client. It assumes that client is being invoked within a Kubernetes cluster.
+func InCluster() (Interface, error) {
+
+}
+
+// Client is a concrete implementation of Kubernetes Çlient interface.
+type Client struct {
+}
+
+// Create saves the object obj in to the Kubernetes cluster. obj must have
```

Comment:
If you don't expose the client, then these comments belong to the interface, which I think would be better anyway since you're not really describing the specifics of that particular client, but the interface.

Reply:
Totally makes sense. Done.